app/src/main/java/com/mstudio/superheroarch/presentation/detail/CharacterDetailViewModel.kt

```diff
@@ -4,27 +4,29 @@ import com.mstudio.superheroarch.presentation.model.CharacterData
 import com.mstudio.superheroarch.presentation.network.NetworkManager
 import com.mstudio.superheroarch.usecase.CharacterAndEpisodeData
 import com.mstudio.superheroarch.usecase.GetCharacterAndEpisodeUseCase
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
 class CharacterDetailViewModel(
-    private val view: CharacterDetailViewTranslator
+    private val view: CharacterDetailViewTranslator,
+    private val useCase: GetCharacterAndEpisodeUseCase = GetCharacterAndEpisodeUseCase(),
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO,
+    private val networkManager: NetworkManager = NetworkManager()
 ) {
```

Comment:
try to extend from androidx.lifecycle.ViewModel, then you can use viewModelScope instead of CoroutineScope, and you'll be able to use the Uncofined in testing.